src/helper_lib/__init__.py

```diff
@@ -0,0 +1,178 @@
+# This program and the accompanying materials are made available under the
+# terms of the Apache License Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+
+import os
+import subprocess
+from pathlib import Path
+from sphinx_needs.logging import get_logger
+
+LOGGER = get_logger(__name__)
+
+
+def find_ws_root() -> Path | None:
+    """
+     Find the current MODULE.bazel file
+
+    Please note that this does behave differently depending on which command you use:
+    'bazel run' => Path
+    'bazel build' => None
+    'esbonio / direct execution' => None
+    """
+    ws_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", None)
+    return Path(ws_dir) if ws_dir else None
+
+
+def find_git_root(start_path: str | Path = "") -> Path | None:
+    """
+    Find the git root directory starting from the given path or __file__.
+
+    Please note that this does behave differently depending on which command you use:
+    'bazel run' => Path
+    'bazel build' => None
+    'esbonio / direct execution' => Path (if executed without input)
+    """
+    if start_path == "":
+        start_path = __file__
+
+    git_root = Path(start_path).resolve()
+    # Note: recoursion_break is a variable that will stop
+    #       certain execution types, like esbonio
+    #       from recousing forever if it can't find a .git
+    #       in the 'home' or 'root' folder.
+    recoursion_break = False
+    while not (git_root / ".git").exists():
+        git_root = git_root.parent
+        if git_root == Path("/"):
+            # fallback to cwd when building with python -m sphinx docs _build -T
+            if recoursion_break:
+                return None
+            git_root = Path.cwd().resolve()
+            recoursion_break = True
+    return git_root
+
+
+def parse_remote_git_output(str_line: str) -> str:
+    """
+    This function parses the output from get_github_repo_info
+    and returns <user/org>/<repository>
+
+    Example:
+        Input:
+            => origin git@github.com:MaximilianSoerenPollak/docs-as-code.git
+        Output:
+            => MaximilianSoerenPollak/docs-as-code
+    """
+    if len(str_line.split()) < 2:
+        LOGGER.warning(
+            f"Got wrong input line from 'get_github_repo_info'. Input: {str_line}. "
+            + "Expected example: 'origin git@github.com:user/repo.git'"
+        )
+        return ""
+    url = str_line.split()[1]  # Get the URL part
+    # Handle SSH format (git@github.com:user/repo.git)      Get part after github.com/
+    path = url.split(":")[1] if url.startswith("git@") else "/".join(url.split("/")[3:])
+    return path.replace(".git", "")
+
+
+def get_github_repo_info(git_root_cwd: Path) -> str:
+    """
+    Unlike other funcions, this function behaves the same for the following commands:
+
+    'bazel run' => str
+    'bazel build' => str
```

Comment:
bazel build should not have .git in the sandbox? And it could run remotely where there is definitely no .git.